docs/src/modules/components/AppSearch.js

```diff
@@ -6,62 +6,8 @@ import useMediaQuery from '@material-ui/core/useMediaQuery';
 import { fade, useTheme, makeStyles } from '@material-ui/core/styles';
 import Input from '@material-ui/core/Input';
 import SearchIcon from '@material-ui/icons/Search';
-import loadScript from 'docs/src/modules/utils/loadScript';
 import { handleEvent } from 'docs/src/modules/components/MarkdownLinks';
-
-let searchTimer;
-let initialized = false;
-let dependenciesLoaded = false;
-
-function loadDependencies() {
-  if (dependenciesLoaded) {
-    return;
-  }
-  dependenciesLoaded = true;
-
-  loadCSS(
-    'https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css',
-    document.querySelector('#app-search'),
-  );
-  loadScript(
-    'https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js',
-    document.querySelector('head'),
-  );
-}
-
-function initDocsearch(userLanguage) {
-  clearInterval(searchTimer);
-  searchTimer = setInterval(() => {
-    const docsearchInput = document.querySelector('#docsearch-input');
-
-    if (!window.docsearch || !docsearchInput) {
-      return;
-    }
-
-    clearInterval(searchTimer);
-
-    if (initialized === docsearchInput) {
-      return;
-    }
-
-    initialized = docsearchInput;
-    window.docsearch({
-      apiKey: '1d8534f83b9b0cfea8f16498d19fbcab',
-      indexName: 'material-ui',
-      inputSelector: '#docsearch-input',
-      algoliaOptions: {
-        facetFilters: ['version:master', `language:${userLanguage}`],
-      },
-      handleSelected: (input, event, suggestion) => {
-        event.button = 0;
-        const parseUrl = url.parse(suggestion.url);
-        handleEvent(event, parseUrl.pathname + parseUrl.hash);
-        input.close();
-      },
-      // debug: true, // Set debug to true if you want to inspect the dropdown.
-    });
-  }, 100);
-}
+import docsearch from 'docsearch.js';
```

Comment:
Lol, nobody seems to use the same strategy:
- React, loads JS from unpkg, loads no CSS: https://github.com/reactjs/reactjs.org/blob/06a8850b2f92908f6056f161b28c716dcdedddac/src/html.js#L4
- Gatsby, loads JS from a local copy, loads CSS from the CDN: https://github.com/gatsbyjs/gatsby/blob/194a97c3854e3b8e4fcdd5b1eab1d72450dea867/www/src/components/search-form.js#L18
- Storybook, loads JS async from an effect, loads CSS from the CDN: https://github.com/styled-components/styled-components-website/blob/e0c9a10617122315ba56704b75ddb7d8ed653c86/components/Nav/SearchWithAlgolia.js#L12
- Facebook docs, loads JS async from an effect, loads CSS from the CDN: https://github.com/facebook/docusaurus/blob/0bbf7b8202963e09f7fe2ce7f23077018eb5fdbd/packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js#L36

I have run a diff test: https://www.webpagetest.org/video/compare.php?tests=190916_BW_dfe76371d3034b92b3c5845958e779a5,190916_AK_e08762707126c2b3561018291021f807. . Seems to be the same.

Reply:
lol